node_app/models/Patient.js

```diff
@@ -0,0 +1,188 @@
+  address: {
+    use: enum (home, work, temp)
+    line: string
+    city: string
+    state: string
+    postalCode: string
+    country: string
+  }
+  contact: {
+    family: string
+    given: string
+    phone: {
+      value: string
+      use: enum (home, work, temp, old, mobile)
+    }
+  }
+  language: enum (English, Spanish),
+  relationship: enum (DOMPART Domestic Partner, INLAW In-Law, CHILD Child, CHLDFOST Foster Child, SPS Spouse, PRN Parent, GRPRN Grandparent, O Other, ONESELF)
+}
+
+{
+  "Patient": [
+    {
+      "family": "Smith",
+      "given": "John",
+      "suffix": "Jr.",
+      "phone": [
+        {
+          "value": "(000)000-000",
+          "use": "mobile"
+        },
+        {
+          "value": "(100)000-000",
+          "use": "mobile"
+        }
+      ],
+      "email": [
+        "email@site.com",
+        "email2@site.com"
+      ],
+      "gender": "male",
+      "birthDate": "2000-01-01",
+      "race": "",
+      "ethnicity": "",
+      "address": {
+        "use": "home",
+        "line": "90 Walnut St",
+        "city": "New York City",
+        "state": "NY",
+        "postalCode": "14623",
+        "country": "USA"
+      },
+      "contact": {
+        "family": "Smith",
+        "given": "John",
+        "phone": {
+          "value": "(110)000-000",
+          "use": "mobile"
+        }
+      },
+      "language": "English",
+      "relationship": "ONESELF"
+    }
+  ]
+}
+
+*/
+exports.toFHIR = function (patient) {
+  let resource = {
+    resourceType: "Patient",
+    name: [
+      {
+        family: patient.family,
+        given: [patient.given],
+        // middle and suffix are optional
+      },
+    ],
+    telecom: [
+      // add later
+    ],
+    gender: patient.gender,
+    birthDate: patient.birthDate,
+    address: [
+      {
+        use: patient.address.use,
+        line: [patient.address.line],
+        city: patient.address.city,
+        state: patient.address.state,
+        postalCode: patient.address.postalCode,
+        country: patient.address.country,
+      },
+    ],
+    photo: [
+      {
+        url: "", // add later
+        title: "Photo of user",
+      },
+    ],
+    contact: [
+      {
+        relationship: [
+          {
+            coding: [
+              {
+                system: "http://terminology.hl7.org/CodeSystem/v2-0131",
+                code: "C",
+                display: "Emergency Contact",
+              },
+            ],
+          },
+        ],
+        name: {
+          family: patient.contact.family,
+          given: [patient.contact.given],
+        },
+        telecom: [
+          {
+            system: "phone",
+            value: patient.contact.phone.value,
+            use: patient.contact.phone.use,
+          },
+        ],
+      },
+    ],
+    communication: [
+      {
+        language: {
+          text: patient.language,
+        },
+        preferred: true,
+      },
+    ],
+    link: [
```

Comment:
Can we take this out now that link handling is at the bottom of this function?

Reply:
No, the field needs to be initialized for use later